webpack.config.js

```diff
@@ -18,37 +18,45 @@ const { getIfUtils, removeEmpty } = require( 'webpack-config-utils' );
 
 
 module.exports = ( env ) => {
-  const { ifProd, ifNotProd, ifTest, ifNotTest } = getIfUtils( env );
+  const { ifProd, ifNotProd, ifTest, ifDev, ifSite } = getIfUtils( env, ['prod', 'test', 'dev', 'site'] );
+
+  function _packagePrefix() {
+    return './packages' + ( env.element ? `/${env.element}` : '' );
+  }
+
+  function _entryPoint() {
+    if ( ifTest() ) {
+
+      return {
+        'test': _packagePrefix() + '/index.test.ts',
+        'test-helpers' : './test-helpers.ts'
+      };
+
+    }
+
+    if ( ifProd() ) {
+
+      return {
+        'index': _packagePrefix() + '/index.ts',
```

Comment:
**NOTE** 🚨

this build entry points are not doing what you think they are doing.

`index.ts` imports `define` from skate, so as a end result. skate will be always bundled. This is not what we wanted.

What you need to do:

- for prod build you have to provide `externals` for main index.ts packages 
```
externals: {
  skate: 'skatejs'
}
```
- for prod build of `index-with-deps.ts` that vendor import is completely unnecessary and may increase bundle size. We are using three shaking, so only used functions/imports are bundled. only thing that you need to import there are `polyfills`